src/lib/mina/actions/offchain-contract-tests/ExampleContract.ts

```diff
@@ -0,0 +1,89 @@
+import {
+  SmartContract,
+  method,
+  state,
+  PublicKey,
+  UInt64,
+  Experimental,
+} from '../../../../index.js';
+
+export { offchainState, StateProof, ExampleContract };
+
+const { OffchainState } = Experimental;
+
+const offchainState = OffchainState(
+  {
+    accounts: OffchainState.Map(PublicKey, UInt64),
+    totalSupply: OffchainState.Field(UInt64),
+  },
+  { logTotalCapacity: 10, maxActionsPerProof: 5 }
+);
+
+class StateProof extends offchainState.Proof {}
+
+// example contract that interacts with offchain state
+class ExampleContract extends SmartContract {
+  @state(OffchainState.Commitments) offchainStateCommitments =
+    offchainState.emptyCommitments();
+
+  // o1js memoizes the offchain state by contract address so that this pattern works
+  offchainState: any = offchainState.init(this);
```

Comment:
`any` is required here because the type of `offchainState` is actually

```
OffchainStateInstance<{
    readonly accounts: { kind: "offchain-map"; keyType: typeof PublicKey; valueType: typeof UInt64; };
    readonly totalSupply: { kind: "offchain-field"; type: typeof UInt64; };
  }>
```

But that type is not exported, resulting in this error: `Public property 'offchainState' of exported class has or is using name 'OffchainStateInstance' from external module "/Users/coby/Projects/o1js/src/lib/mina/actions/offchain-state" but cannot be named.`

Reply:
Seems fine to me to handle in a different PR eventually, as this API remains experimental.

Reply:
as a rule of thumb: If we can do it now, might as well do it now :D

Reply:
woah doesn't omitting this export completely break the type-safety of interacting with offchain state then?

Reply:
Turns out to be a pretty simple fix

https://github.com/o1-labs/o1js/pull/1871